src/cargo/core/compiler/compilation.rs

```diff
@@ -248,6 +248,7 @@ impl<'cfg> Compilation<'cfg> {
         kind: CompileKind,
         pkg: &Package,
         script_meta: Option<Metadata>,
+        jobserver_client: Option<&jobserver::Client>,
```

Comment:
I'm not clear, why is this being added as a new parameter if it is always passed `config.jobserver_from_env()`? Couldn't `target_process` just call `self.config.jobserver_from_env()` itself?

Reply:
You're absolutely correct. I didn't realize it has the access to `config` 🙇🏾‍♂️.

Addressed.